### PR TITLE
Wire in dense gather.

### PR DESCRIPTION
**Pull request by dlibenzi**

### CODE REVIEW

File: torch_xla/csrc/ops/gather.cpp [hunk 1]
```diff
@@ -11,11 +12,24 @@ namespace ir {
 namespace ops {
 namespace {
 
+bool IsSparseGather(const xla::XlaOp& input, const xla::XlaOp& index,
+                    xla::int64 dim) {
+  xla::Shape input_shape = XlaHelpers::ShapeOfXlaOp(input);
+  xla::Shape index_shape = XlaHelpers::ShapeOfXlaOp(index);
+  xla::int64 input_elements = xla::ShapeUtil::ElementsIn(input_shape);
```
Comment (taylanbil) [inline torch_xla/csrc/ops/gather.cpp]: this seems like it can have its own util function like [tensor -> num_elements]

**Comment by taylanbil** [comment]
I'm not familiar w/ dense/sparse gather, but this lgtm. Did you test?

File: torch_xla/csrc/ops/gather.cpp (hunk 1 above)
Comment (dlibenzi) [reply torch_xla/csrc/ops/gather.cpp]: It is not as widely used to have an API IMHO.
The other issue is that getting a Shape of an XLA op is a bit expensive, so I fear folks would use the new API, and then get the Shape in order to access more of its data, effectively materializing it twice.

**Comment by dlibenzi** [comment]
> I'm not familiar w/ dense/sparse gather, but this lgtm. Did you test?

Yes, tests are green on my machine.
The code is here:

https://github.com/tensorflow/tensorflow/blob/13e631e9f7b4e70f50c58e961b56bb9e81fe1ff3/tensorflow/compiler/xla/client/lib/slicing.cc#L143

**Pull request merged**